### PR TITLE
Add support for google_compute_region_network_endpoint gcp resource

### PR DESCRIPTION
**Pull request by JonathanO**
### Description of your changes

This change adds support for the [google_compute_region_network_endpoint](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_region_network_endpoint) resource to the GCP provider.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Tested in a local kind cluster

### CODE REVIEW

**Comment by jeanduplessis** [comment]
/test-examples="examples/compute/regionnetworkendpoint.yaml"

**Comment by jeanduplessis** [comment]
/test-examples="examples/compute/regionnetworkendpoint.yaml"

File: config/externalname.go [hunk 1]
```diff
@@ -236,6 +236,8 @@ var terraformPluginSDKExternalNameConfigs = map[string]config.ExternalName{
 	"google_compute_region_health_check": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/regions/{{ .parameters.region }}/healthChecks/{{ .external_name }}"),
 	// Imported by using the following format: projects/{{project}}/regions/{{region}}/networkEndpointGroups/{{name}}
 	"google_compute_region_network_endpoint_group": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/regions/{{ .parameters.region }}/networkEndpointGroups/{{ .external_name }}"),
+	// Imported by using the following format: projects/{{project}}/regions/{{region}}/networkEndpointGroups/{{name}}
```
Comment (turkenf) [inline config/externalname.go:239]: I think this comment line is for `networkEndpointGroup` resource, could you please change it with the following?

```suggestion
	// Imported by using the following format: projects/{{project}}/regions/{{region}}/networkEndpointGroups/{{region_network_endpoint_group}}/{{ip_address}}/{{fqdn}}/{{port}}
```

**Review by turkenf — COMMENTED** [review]
Many thanks for your contribution @JonathanO, left a small comment for you to consider.

**Review by turkenf — APPROVED** [review]

**Pull request merged**